File: Eulerjava/Euler3.java
public class Euler3{
    public static void main(String[] args){
	long num = 600851475143L;
	int large = 0;
	for(int i = 3; i < (int)Math.sqrt(num); i++){
	    if(num%i == 0 && isprime(i)){
		    large = i;
		}
	}
	    System.out.println(large);
    }

    public static boolean isprime(int n){
	if(n%2 == 0){
	    return false;
	}
	for(int i = 3; i < (int)Math.sqrt(n) + 1; i++){
	    if(n%i == 0){
		return false;
	    }
	}
	return true;
    }
}

File: Eulerc/Euler3.c

#include<stdio.h>
#include<math.h>
//needed to use -lm to end
int isprime(long a){
	double x = (double) a;
	int i;
	if(((int) a)%2==0)
		return 0;
	for (i = 3 ; i <= a; i+=2){
		if((int)a%i == 0){
			return 0;
		}
	}
	return 1;

}


int main (void){
	long j = 600851475143;
	int stuff= 0;
	long i;
	for (i=3; i <=j/2; i+=2){
		if( (j%i == 0) && isprime(i) ){
			stuff = i;
		}
	}
	printf("%i\n",stuff);
	return 0;
}
